onnxruntime/core/providers/nnapi/nnapi_builtin/builders/model_builder.cc

```diff
@@ -515,7 +522,37 @@ Status ModelBuilder::Compile(std::unique_ptr<Model>& model) {
       nnapi_->ANeuralNetworksModel_finish(nnapi_model_->model_),
       "on model finish");
 
+  // We have a list of target devices, try to see if the model can be run entirely
+  // using the list of target devices
+  // This is only available on API 29+, for API 28- the nnapi_target_devices_ will
+  // be empty so we will not check API level here, see GetTargetDevices()
+  bool use_create_for_devices = false;
   if (!nnapi_target_devices_.empty()) {
+    std::unique_ptr<bool[]> supported_ops_holder = onnxruntime::make_unique<bool[]>(num_nnapi_ops_);
+    auto* supported_ops = supported_ops_holder.get();
+    RETURN_STATUS_ON_ERROR_WITH_NOTE(
+        nnapi_->ANeuralNetworksModel_getSupportedOperationsForDevices(
+            nnapi_model_->model_, nnapi_target_devices_.data(),
+            nnapi_target_devices_.size(), supported_ops),
+        "on getSupportedOperationsForDevices");
+
+    auto* it = std::find(supported_ops, supported_ops + num_nnapi_ops_, false);
+    if (it != supported_ops + num_nnapi_ops_) {
+      // There are some ops not supported by the list of the target devices
```

Comment:
nit: std::any_of may be slightly clearer

Reply:
Change to std::all_of